src/odemis/driver/technolution.py

```diff
@@ -518,45 +438,136 @@ def __init__(self, name, role, parent, **kwargs):
+        """
+        # The calibration frequency is the inverse of the total line scan time.
+        calibration_frequency = 1 / total_line_scan_time
+
+        # Calculate the sampling period and number of setpoints (the sampling period in seconds is not needed here)
+        tick_calibration_dwell_time, _, nmbr_scanner_points = self._calc_calibration_sampling_period(
+                                                                        total_line_scan_time,
+                                                                        self.parent.clockPeriod.value,
+                                                                        self.parent._mirror_descanner.clockPeriod.value)
+
+        # Determine the amplitude of the setpoints function in bits.
+        scan_gain_volts = convertRange(self.scanGain.value, numpy.array(self.scanGain.range)[:, 1], VOLT_RANGE)
+        scan_offset_volts = convertRange(self.scanOffset.value, numpy.array(self.scanOffset.range)[:, 1], VOLT_RANGE)
+        scan_amplitude_volts = scan_gain_volts - scan_offset_volts
+
+        # Using the time stamps the setpoints are determined.
+        time_points_scanner = numpy.linspace(0, total_line_scan_time, nmbr_scanner_points)
+        x_setpoints = scan_offset_volts[0] + scan_amplitude_volts[0] * numpy.sin(
+                2 * math.pi * calibration_frequency * time_points_scanner)
+        y_setpoints = scan_offset_volts[1] + scan_amplitude_volts[1] * signal.sawtooth(
+                2 * math.pi * calibration_frequency * time_points_scanner)
+
+        return x_setpoints.tolist(),\
+               y_setpoints.tolist(),\
+               tick_calibration_dwell_time
+
+    def _calc_calibration_sampling_period(self, total_line_scan_time, system_clock_period, descanner_clock_period):
+        """
+        For creation of the calibration setpoints the sampling period is variable but the total number of
+        setpoints is limited to maximum 9000 points. Depending on the total line scanning time the calibration
+        setpoints have a variable sampling period (calibration dwell time) which is a multiple of the system
+        clock period. The number of setpoints (and the corresponding sampling period is chosen such that the
+        biggest number of setpoints is used (highest resolution of the signal).
+        The sampling period is defined by finding the higest number of setpoints for the calibration signal
+        with a sampling period equal to a whole multiple of the system clock period. This is done by iteratively
+        increasing the sampling period.
+
+        :param total_line_scan_time (float): Total line scanning time in seconds
+        :param system_clock_period (float): clock period of the system and scanner in seconds
+        :param descanner_clock_period (float): Descanner lock period of the scanner in seconds
+        :return: (int) Sampling period in ticks
+                 (float) Sampling period is seconds (not used yet)
```

Comment:
I would remove "not used yet" here. You already mention it in the caller and also here in the method in the return.